packages/react/src/components/Menu/MenuItem.tsx

```diff
@@ -342,31 +347,40 @@ class MenuItem extends AutoControlledComponent<ReactProps<MenuItemProps>, MenuIt
     _.invoke(this.props, 'onFocus', e, this.props)
   }
 
-  private closeAllMenus = (e, focusNextParent: boolean) => {
-    const { menu, inSubmenu } = this.props
+  private isSubmenuOpen = (): boolean => {
+    const { menu } = this.props
     const { menuOpen } = this.state
-    if (menu && menuOpen) {
-      this.trySetMenuOpen(false, e, () => {
-        if (!inSubmenu && (!focusNextParent || this.props.vertical)) {
-          focusAsync(this.itemRef.current)
-        }
-      })
+
+    return menu && menuOpen
```

Comment:
maybe enforce boolean here? if menu is undefined, this will not return false.

Reply:
enforcing typing is good, but I can't come up with a case where it can turn into a problem. Do you have a concrete example on your mind? Just to make sure if I really need to rewrite it to
`return menu && menuOpen ? true : false`

Reply:
I'd say it's fine, we care about truthy / nontruthy values only in this case

Reply:
I did not look at the workflow, just at the signature saying that it returns a `boolean`. If it's not relevant and it can return the `undefined` safely then leave it, otherwise `return !!(menu && menuOpen)`.

Reply:
it is safe for that case to have `undefined` however would agree with you on consistency of the returning type. Will apply your suggestion. Thank you!